app/Http/Controllers/BeatmapsController.php

```diff
@@ -69,30 +69,28 @@ public function scores($id)
                 $query = $model
                     ->where('beatmap_id', $beatmap->getKey())
                     ->with(['beatmap', 'user.country'])
-                    ->defaultListing();
+                    ->withMods($mods)
+                    ->withType($type, compact('user'));
 
-                $query->withMods($mods);
-                $query->withType($type, compact('user'));
+                if ($user !== null) {
+                    $score = (clone $query)->where('user_id', $user->user_id)->first();
+                }
 
                 $results = [
-                    'scores' => json_collection($query->forListing(), 'Score', ['beatmap', 'user', 'user.country']),
+                    'scores' => json_collection($query->defaultListing()->forListing(), 'Score', ['beatmap', 'user', 'user.country']),
                 ];
-            } catch (ScoreRetrievalException $ex) {
-                return error_popup($ex->getMessage());
-            }
-
-            if ($user !== null) {
-                $score = (clone $query)->where('user_id', $user->user_id)->first();
 
-                if ($score !== null) {
+                if (isset($score)) {
                     $results['userScore'] = [
                         'position' => $score->userRank(compact('type', 'mods')),
                         'score' => json_item($score, 'Score', ['user', 'user.country']),
                     ];
```

Comment:
@peppy @smoogipoo need some confirmation if lazer can handle `null` position and if there's any difference if `userScore` is not set versus `userScore: null` for a beatmap's scores

Reply:
not being present will result in a position of 0. null will require a change at our end to support it.

Reply:
we're going to go with not setting `position` if it's not available for now